### DISCUSSION TITLE
Sync specific table to the server.

### DISCUSSION THREAD

**Discussion by PabloRuiz-TT** [Azure/azure-mobile-apps#660] [Q&A] [answered]
I’ve been following the documentation for the offline sync, I found the way to sync specific tables to the server by creating an array with the tables name that we want to sync inside the RefreshItemsAsync method, I am using the following code:

public async Task RefreshItemsAsync()
{
    await InitializeAsync();
    var tableToPush = new string[] = { "Account" };
    await _client.PushTablesAsync(tablesToPush);

   await _offlineTable.PushItemsAsync();
   return;
}

But when I create a record in the local database, it is also created in the azure database. Is there something am I missing or is this the right way to do it?

**Comment by adrianhall** [comment]
Are you pulling those tables somewhere?  There is an implicit push when you do a pull.

**Reply by PabloRuiz-TT** [reply]
The following configuration is supposed to override the implicit PushAsync(), example:
`PullOptions pullOptions = new() { PushOtherTables = false };`
`await _offlineTable.PullItemsAsync(pullOptions);`

Is this correct?

**Reply by adrianhall** [reply, marked as answer]
Correct.  To really do what you would need, you would want the following:

You need something like:

``` csharp
string[] tablesToSync = new string[] { "table1", "table2" };
PullOptions pullOptions = new() { PushOtherTables = false }; // this is actually the default
await client.PushTablesAsync(tablesToSync);
foreach (var tableName in tablesToSync) {
  IOfflineTable tableRef = client.GetOfflineTable(tableName);
  await tableRef.PullTablesAsync(pullOptions);
}
```
